src/Ormolu/Utils/Cabal.hs

```diff
@@ -59,20 +59,18 @@ defaultCabalInfo =
       ciCabalFilePath = Nothing
     }
 
--- | Locate .cabal file corresponding to the given Haskell source file and
--- obtain 'CabalInfo' from it.
+-- | Locate a @.cabal@ file corresponding to the given Haskell source file
+-- and obtain 'CabalInfo' from it.
 getCabalInfoForSourceFile ::
   (MonadIO m) =>
   -- | Haskell source file
   FilePath ->
-  -- | Extracted cabal info
-  m CabalInfo
+  -- | Extracted cabal info, if any
+  m (Maybe CabalInfo)
 getCabalInfoForSourceFile sourceFile = liftIO $ do
   findCabalFile sourceFile >>= \case
-    Just cabalFile -> parseCabalInfo cabalFile sourceFile
-    Nothing -> do
-      hPutStrLn stderr $ "Could not find a .cabal file for " <> sourceFile
```

Comment:
There is another place where we log to stderr in this module:

https://github.com/tweag/ormolu/blob/5a43ffc288df4ebce718340da5e7586931de61db/src/Ormolu/Utils/Cabal.hs#L147-L151

Now the question is whether we want to indicate in the return type whether we couldn't find a `.cabal` file or whether it didn't mention the source file...

Reply:
I think we need to introduce a type that would describe the exact outcome:

```haskell
data CabalSearchResult = CabalNotFound | DidNotMention | CabalFound CabalInfo
```

And then we can do printing as required in `Main.hs` and erase this info by using `defaultCabalInfo` when necessary.

Reply:
Yeah, that sounds good, potentially also with a `FilePath` field for `DidNotMention` containing the `.cabal` file.

The alternative would be to pass in some "tracer" (i.e. a wrapped `String -> IO ()` or even `CabalSearchResult -> IO ()`), which also seems reasonable to me.